service/history/historyEngine.go

```diff
@@ -690,108 +698,173 @@ func (e *historyEngineImpl) QueryWorkflow(
 	ctx ctx.Context,
 	request *h.QueryWorkflowRequest,
 ) (retResp *h.QueryWorkflowResponse, retErr error) {
-	domainCache, err := e.shard.GetDomainCache().GetDomainByID(request.GetDomainUUID())
+
+	context, release, err := e.historyCache.getOrCreateWorkflowExecution(ctx, request.GetDomainUUID(), *request.GetRequest().GetExecution())
```

Comment:
plz put this function logic into a dedicated file, history engine is already too large

Reply:
As we spoke about offline this should be done as part of a larger refactor where all the logics in historyEngine are split into different files.